cmd/thanos/receive.go

```diff
@@ -325,10 +325,10 @@ func runReceive(
 				if httpProbe.IsReady() {
 					minTime, maxTime := mts.TimeRange()
 					return &infopb.StoreInfo{
-						MinTime:           minTime,
-						MaxTime:           maxTime,
-						SupportsSharding:  true,
-						SendsSortedSeries: true,
+						MinTime:                      minTime,
+						MaxTime:                      maxTime,
+						SupportsSharding:             true,
+						SupportsWithoutReplicaLabels: false, // TODO(bwplotka): Add support for efficiency.
```

Comment:
Why not implement this for every store in the same PR? It should involve just calling a single function on the external labels?

Reply:
Yea... "just" and then 1000 lines  to fix and have tests (: Next PR!